sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -142,13 +142,17 @@ object OrcUtils extends Logging {
       reader: Reader,
       conf: Configuration): Option[(Array[Int], Boolean)] = {
     val orcFieldNames = reader.getSchema.getFieldNames.asScala
+    val forcePositionalEvolution = OrcConf.FORCE_POSITIONAL_EVOLUTION.getBoolean(conf)
     if (orcFieldNames.isEmpty) {
       // SPARK-8501: Some old empty ORC files always have an empty schema stored in their footer.
       None
     } else {
-      if (orcFieldNames.forall(_.startsWith("_col"))) {
-        // This is a ORC file written by Hive, no field names in the physical schema, assume the
-        // physical schema maps to the data scheme by index.
+      if (forcePositionalEvolution || orcFieldNames.forall(_.startsWith("_col"))) {
+        // This is either an ORC file written by an old version of Hive and there are no field
+        // names in the physical schema, or a file written by a newer version of Hive where
+        // `orc.force.positional.evolution=true` (possibly because columns were renamed so the
```

Comment:
> a file written by a newer version of Hive where `orc.force.positional.evolution=true`

Does it read correctly? It reaches here just because users enable the config, right? I don't see we check if the file is written `orc.force.positional.evolution=true`.

Reply:
Thanks, fixed it in https://github.com/apache/spark/pull/29737/commits/a074b63fb240c62c7fd09aca1cdc66300b36b2dc